### PR TITLE
Qwebengineview flashing workaround

### PR DESCRIPTION
**Pull request by w0lek**
> ### Motivate of the pull request
> - [ ] To address an existing issue. If so, please provide a link to the issue: <issue id>
> - [ ] Breaking new feature. If so, please describe details in the description part.

> ### Describe the technical details
> #### What is currently done? (Provide issue link if applicable)
> 
> 
> 
> 
> 
>
> #### What does this pull request change?
> 
> 
> 
> 
> 

> ### Which part of the code base require a change
> 
> - [ ] Library: <Specify the library name>
> - [ ] Plug-in: <Specify the plugin name>
> - [ ] Engine
> - [ ] Documentation
> - [ ] Regression tests
> - [ ] Continous Integration (CI) scripts

> ### Impact of the pull request

> - [ ] Require a change on Quality of Results (QoR)
> - [ ] Break back-compatibility. If so, please list who may be influenced.

### CODE REVIEW

**Comment by w0lek** [comment]
@coolbreeze413 @volodymyrkochyn @ravic-rs 
the workaround didn't fix issue fully.
@coolbreeze413 sorry i was wrong in teams, more infor below.
We still see small blinking on welcomepage first showup. This maybe not noticeable from first glance. 
So basically i was able to move flashing, from a moment of showing Editor, to a moment when welcome page first time shown up.

Note:
The flashing issue doesn't depend on QTabWidget, it's reproducible with QWidget+QLayout as well, and pure QWidget as a parent to QWebEngineView. So refactoring QTabWidget, as i suggested earlier will not work for fixing this issue.

Note2:
The main problem as i understood, we add QWebEngineView on already visible widget. In Qt examples, when we custruct UI and add QWebView before any widget was shown, the flashing doesn't occur. 
Hovever, it seems this isn't a case for FOEDAG, where workaround actually doing the same (it adds the QWebENgineView before showEvent). On FOEDAG we still see small blinking on startup. 
If we really annoying by this startup blinking, we may probably investigate this issue in more deep.
May you give a try and say your opinion?

File: src/Main/CMakeLists.txt [hunk 1]
```diff
@@ -23,6 +23,11 @@ cmake_minimum_required(VERSION 3.15)
 
 project(foedagcore LANGUAGES CXX)
 
+if(${FOEDAG_WITH_MONACO_EDITOR})
+  message("using FOEDAG_WITH_MONACO_EDITOR_WORKAROUND_FOR_QWEBENGINEVIEW_FLASHING")
+  add_definitions(-DFOEDAG_WITH_MONACO_EDITOR_WORKAROUND_FOR_QWEBENGINEVIEW_FLASHING)
```
Comment (KochynVolodymyr) [inline src/Main/CMakeLists.txt:28]: we don't need additional flag for that

Comment (w0lek) [reply src/Main/CMakeLists.txt]: i have corrected this

File: src/Main/CMakeLists.txt [hunk 2]
```diff
@@ -152,7 +152,14 @@ else()
   set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -Wno-deprecated-declarations -Werror")
 endif()
 
-target_link_libraries(foedagcore PUBLIC  Qt6::Widgets Qt6::Core Qt6::Gui Qt6::Xml compiler)
+target_link_libraries(foedagcore PUBLIC Qt6::Widgets Qt6::Core Qt6::Gui Qt6::Xml compiler)
+if(FOEDAG_WITH_MONACO_EDITOR)
+  target_link_libraries(foedagcore PUBLIC Qt6::WebEngineWidgets)
+endif()
+if (FOEDAG_WITH_MONACO_EDITOR)
```
Comment (KochynVolodymyr) [inline src/Main/CMakeLists.txt:159]: if(FOEDAG_WITH_MONACO_EDITOR)
  target_link_libraries(foedagcore PUBLIC Qt6::WebEngineWidgets)
  target_compile_definitions(foedagcore PUBLIC FOEDAG_WITH_MONACO_EDITOR=${FOEDAG_WITH_MONACO_EDITOR})
endif()

File: src/MainWindow/main_window.cpp [hunk 3]
```diff
@@ -123,6 +126,28 @@ void centerWidget(QWidget& widget) {
...
+   temporary QWebEngineView with zero size into the main window, and then delete
+   it after some time (in post showEvent period), since it doesn't play any
+   specific role, except the role of having it for initialization.
+  */
+  const QString preloadWebViewName{"webViewPreloader"};
+  QWebEngineView* preloadWebView = new QWebEngineView(this);
+  preloadWebView->resize(0, 0);
+  preloadWebView->setObjectName(preloadWebViewName);
+  // remove event must take place after showEvent finish it's routing
+  QTimer::singleShot(1000, [this, preloadWebViewName]() {
```
Comment (KochynVolodymyr) [inline src/MainWindow/main_window.cpp:143]:
```
QTimer::singleShot(1000, [preloadWebView]() {
      preloadWebView->deleteLater();
  });
```

**Review by KochynVolodymyr — APPROVED** [review]
Looks good. I don't see flickering issue

**Pull request merged**